app/routes.py

```diff
@@ -1,2 +1,263 @@
-from flask import Blueprint
+from flask.wrappers import Response
+from app import db
+from app.models.task import Task, to_dict, to_dict_goal
+from app.models.goal import Goal, to_json
+from flask import Blueprint, request, make_response, jsonify
+from datetime import datetime
+from dotenv import load_dotenv
+import os
+import requests
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+load_dotenv()
+
+# tasks
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+
+    request_body = request.get_json()
+
+    response = {"details": "Invalid data"}
+
+    if "title" not in request_body.keys() or "description" not in request_body.keys() or "completed_at" not in request_body.keys():
+
+        return jsonify(response), 400
+
+    else:
+        new_task = Task(title = request_body["title"], description = request_body["description"], completed_at = request_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+        valid_task = {"task": to_dict(new_task)}
+
+        return jsonify(valid_task), 201
+
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_tasks():
+
+    tasks_response = []
+
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+
+    else:
+        tasks = Task.query.all()
+
+    for task in tasks:
+        tasks_response.append(to_dict(task))
+
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def handle_task(task_id):
+
+    task = Task.query.get(task_id)
+
+    if request.method == "GET":
+        if task is None:
+            return make_response(f"404 Not Found", 404) 
+
+        else:
+            one_task = to_dict(task)
```

Comment:
Here is where the last test fails - if the task has a valid goal_id, it should include the goal_id in the json output.  One way to handle this is:
```
if task.goal_id:
    one_task = to_dict_goal(task)
else:
    one_task = to_dict(task)
```